dfinity/nns-dapp
# Make snsTotalTokenSupplyStore a derived store

# Motivation

Currently, `snsTotalTokenSupplyStore` is a Svelte store which is filled at the same time as the `snsAggregatorStore` and based on the same data. This is fragile and it's more robust to make it a derived store so we have 1 source of truth for the aggregator data.

# Changes

1. Change `snsTotalTokenSupplyStore` to a derived store.
2. Remove code to fill `snsTotalTokenSupplyStore`.
3. Change import paths where `snsTotalTokenSupplyStore` is used. 

# Tests

1. Add a parameter to `aggregatorSnsMockWith` to allow setting the total token supply.
2. Change tests to use `setSnsProjects` instead of `snsTotalTokenSupplyStore.setTotalTokenSupplies`, which, being a derived store, can no longer be set directly.
3.  Move `sns-total-token-supply.store.spec.ts` to `sns-total-token-supply.derived.spec.ts`

# Todos

not necessary
